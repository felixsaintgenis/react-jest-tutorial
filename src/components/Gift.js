import React from "react";
import { Component } from "react";
import {Form, FormGroup, FormControl, ControlLabel, Button} from 'react-bootstrap';

class Gift extends Component {
    constructor(){
        super();

        this.state = {
            person: "",
            present:""
        }
    }
    render() {
        return (
            <div>

                <form>
                    <FormGroup>
                    <ControlLabel>Person</ControlLabel>
                        <FormControl className="input-person" onChange= {event => this.setState({person: event.target.value})} />

                        <ControlLabel>Present</ControlLabel>
                        <FormControl className="input-present" onChange= {event => this.setState({present: event.target.value})} />
                        
                    </FormGroup>
                </form>
                <button className="btn-remove" onClick={() => this.props.removeGift(this.props.gift.id)}>Remove gift</button>
            </div>
        );
    }
}

export default Gift;