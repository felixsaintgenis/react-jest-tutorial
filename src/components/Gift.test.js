import React from 'react';
import {shallow} from 'enzyme';
import Gift from './Gift'

describe('Gift', () => {
    const mockRemove = jest.fn()
    const id = 1
    const props = {gift: {id}, removeGift: mockRemove};
    const gift = shallow(<Gift {...props} />);

    it('render properly', () => {
        expect(gift).toMatchSnapshot();
    });

    it('initialize a person and a gift in the `state`', () => {
        expect(gift.state()).toEqual({person : '', present : '' });
    }); 

    describe('When typing into the person input', () => {
        beforeEach(() => {
            gift.find('.input-person').simulate('change', { target : {value: 'uncle'}});

        }); 

        it('update the person in the `state`', () => {
            expect(gift.state().person).toEqual('uncle')
        });
    });

    describe('When typing into the present input', () => {
        beforeEach(() => {
            gift.find('.input-present').simulate('change', { target : {value: 'golf clubs'}});

        }); 

        it('update the present in the `state`', () => {
            expect(gift.state().present).toEqual('golf clubs')
        });
    });

    
    describe('when clicking on the `remove gift` button', () => {
        beforeEach(() => {
            gift.find('.btn-remove').simulate('click');
        });
    
        it('calls the removeItem function', () => {
            expect(mockRemove).toHaveBeenCalledWith(id)
        });
        
    });
    
});
