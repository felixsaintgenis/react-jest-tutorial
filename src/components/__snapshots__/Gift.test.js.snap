// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Gift render properly 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Gift
    gift={
        Object {
            "id": 1,
          }
    }
    removeGift={[Function]}
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <form>
          <FormGroup
                    bsClass="form-group"
          >
                    <ControlLabel
                              bsClass="control-label"
                              srOnly={false}
                    >
                              Person
                    </ControlLabel>
                    <FormControl
                              bsClass="form-control"
                              className="input-person"
                              componentClass="input"
                              onChange={[Function]}
                    />
                    <ControlLabel
                              bsClass="control-label"
                              srOnly={false}
                    >
                              Present
                    </ControlLabel>
                    <FormControl
                              bsClass="form-control"
                              className="input-present"
                              componentClass="input"
                              onChange={[Function]}
                    />
          </FormGroup>
</form>,
        <button
          className="btn-remove"
          onClick={[Function]}
>
          Remove gift
</button>,
      ],
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": <FormGroup
            bsClass="form-group"
>
            <ControlLabel
                        bsClass="control-label"
                        srOnly={false}
            >
                        Person
            </ControlLabel>
            <FormControl
                        bsClass="form-control"
                        className="input-person"
                        componentClass="input"
                        onChange={[Function]}
            />
            <ControlLabel
                        bsClass="control-label"
                        srOnly={false}
            >
                        Present
            </ControlLabel>
            <FormControl
                        bsClass="form-control"
                        className="input-present"
                        componentClass="input"
                        onChange={[Function]}
            />
</FormGroup>,
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "bsClass": "form-group",
            "children": Array [
              <ControlLabel
                bsClass="control-label"
                srOnly={false}
>
                Person
</ControlLabel>,
              <FormControl
                bsClass="form-control"
                className="input-person"
                componentClass="input"
                onChange={[Function]}
/>,
              <ControlLabel
                bsClass="control-label"
                srOnly={false}
>
                Present
</ControlLabel>,
              <FormControl
                bsClass="form-control"
                className="input-present"
                componentClass="input"
                onChange={[Function]}
/>,
            ],
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "bsClass": "control-label",
                "children": "Person",
                "srOnly": false,
              },
              "ref": null,
              "rendered": "Person",
              "type": [Function],
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "bsClass": "form-control",
                "className": "input-person",
                "componentClass": "input",
                "onChange": [Function],
              },
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "bsClass": "control-label",
                "children": "Present",
                "srOnly": false,
              },
              "ref": null,
              "rendered": "Present",
              "type": [Function],
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "bsClass": "form-control",
                "className": "input-present",
                "componentClass": "input",
                "onChange": [Function],
              },
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
          ],
          "type": [Function],
        },
        "type": "form",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "Remove gift",
          "className": "btn-remove",
          "onClick": [Function],
        },
        "ref": null,
        "rendered": "Remove gift",
        "type": "button",
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <form>
            <FormGroup
                        bsClass="form-group"
            >
                        <ControlLabel
                                    bsClass="control-label"
                                    srOnly={false}
                        >
                                    Person
                        </ControlLabel>
                        <FormControl
                                    bsClass="form-control"
                                    className="input-person"
                                    componentClass="input"
                                    onChange={[Function]}
                        />
                        <ControlLabel
                                    bsClass="control-label"
                                    srOnly={false}
                        >
                                    Present
                        </ControlLabel>
                        <FormControl
                                    bsClass="form-control"
                                    className="input-present"
                                    componentClass="input"
                                    onChange={[Function]}
                        />
            </FormGroup>
</form>,
          <button
            className="btn-remove"
            onClick={[Function]}
>
            Remove gift
</button>,
        ],
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": <FormGroup
              bsClass="form-group"
>
              <ControlLabel
                            bsClass="control-label"
                            srOnly={false}
              >
                            Person
              </ControlLabel>
              <FormControl
                            bsClass="form-control"
                            className="input-person"
                            componentClass="input"
                            onChange={[Function]}
              />
              <ControlLabel
                            bsClass="control-label"
                            srOnly={false}
              >
                            Present
              </ControlLabel>
              <FormControl
                            bsClass="form-control"
                            className="input-present"
                            componentClass="input"
                            onChange={[Function]}
              />
</FormGroup>,
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "bsClass": "form-group",
              "children": Array [
                <ControlLabel
                  bsClass="control-label"
                  srOnly={false}
>
                  Person
</ControlLabel>,
                <FormControl
                  bsClass="form-control"
                  className="input-person"
                  componentClass="input"
                  onChange={[Function]}
/>,
                <ControlLabel
                  bsClass="control-label"
                  srOnly={false}
>
                  Present
</ControlLabel>,
                <FormControl
                  bsClass="form-control"
                  className="input-present"
                  componentClass="input"
                  onChange={[Function]}
/>,
              ],
            },
            "ref": null,
            "rendered": Array [
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "class",
                "props": Object {
                  "bsClass": "control-label",
                  "children": "Person",
                  "srOnly": false,
                },
                "ref": null,
                "rendered": "Person",
                "type": [Function],
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "class",
                "props": Object {
                  "bsClass": "form-control",
                  "className": "input-person",
                  "componentClass": "input",
                  "onChange": [Function],
                },
                "ref": null,
                "rendered": null,
                "type": [Function],
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "class",
                "props": Object {
                  "bsClass": "control-label",
                  "children": "Present",
                  "srOnly": false,
                },
                "ref": null,
                "rendered": "Present",
                "type": [Function],
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "class",
                "props": Object {
                  "bsClass": "form-control",
                  "className": "input-present",
                  "componentClass": "input",
                  "onChange": [Function],
                },
                "ref": null,
                "rendered": null,
                "type": [Function],
              },
            ],
            "type": [Function],
          },
          "type": "form",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "Remove gift",
            "className": "btn-remove",
            "onClick": [Function],
          },
          "ref": null,
          "rendered": "Remove gift",
          "type": "button",
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
    "disableLifecycleMethods": true,
  },
}
`;
